primitives/utility/src/lib.rs

```diff
@@ -410,4 +417,138 @@ mod tests {
+		type TestAssetId = u32;
+		type TestBalance = u128;
+		struct TestAssets;
+		impl MatchesFungibles<TestAssetId, TestBalance> for TestAssets {
+			fn matches_fungibles(a: &MultiAsset) -> Result<(TestAssetId, TestBalance), Error> {
+				match a {
+					MultiAsset { fun: Fungible(amount), id: Concrete(_id) } => Ok((1, *amount)),
+					_ => Err(Error::AssetNotHandled),
+				}
+			}
+		}
+		impl fungibles::Inspect<TestAccountId> for TestAssets {
+			type AssetId = TestAssetId;
+			type Balance = TestBalance;
+
+			fn total_issuance(_: Self::AssetId) -> Self::Balance {
+				todo!()
+			}
+
+			fn minimum_balance(_: Self::AssetId) -> Self::Balance {
+				0
+			}
+
+			fn balance(_: Self::AssetId, _: &TestAccountId) -> Self::Balance {
+				todo!()
+			}
+
+			fn reducible_balance(_: Self::AssetId, _: &TestAccountId, _: bool) -> Self::Balance {
+				todo!()
+			}
+
+			fn can_deposit(
+				_: Self::AssetId,
+				_: &TestAccountId,
+				_: Self::Balance,
+				_: bool,
+			) -> DepositConsequence {
+				todo!()
+			}
+
+			fn can_withdraw(
+				_: Self::AssetId,
+				_: &TestAccountId,
+				_: Self::Balance,
+			) -> WithdrawConsequence<Self::Balance> {
+				todo!()
+			}
+
+			fn asset_exists(_: Self::AssetId) -> bool {
+				todo!()
+			}
+		}
+		impl fungibles::Mutate<TestAccountId> for TestAssets {
+			fn mint_into(_: Self::AssetId, _: &TestAccountId, _: Self::Balance) -> DispatchResult {
+				todo!()
+			}
+
+			fn burn_from(
+				_: Self::AssetId,
+				_: &TestAccountId,
+				_: Self::Balance,
+			) -> Result<Self::Balance, DispatchError> {
+				todo!()
+			}
+		}
+		impl fungibles::Transfer<TestAccountId> for TestAssets {
+			fn transfer(
+				_: Self::AssetId,
+				_: &TestAccountId,
+				_: &TestAccountId,
+				_: Self::Balance,
+				_: bool,
+			) -> Result<Self::Balance, DispatchError> {
+				todo!()
+			}
+		}
+		impl fungibles::Unbalanced<TestAccountId> for TestAssets {
+			fn set_balance(
+				_: Self::AssetId,
+				_: &TestAccountId,
+				_: Self::Balance,
+			) -> DispatchResult {
+				todo!()
+			}
+
+			fn set_total_issuance(_: Self::AssetId, _: Self::Balance) {
+				todo!()
+			}
+		}
+
+		struct FeeChargerAssetsHandleRefund;
+		impl ChargeWeightInFungibles<TestAccountId, TestAssets> for FeeChargerAssetsHandleRefund {
+			fn charge_weight_in_fungibles(
+				_: <TestAssets as Inspect<TestAccountId>>::AssetId,
+				_: Weight,
+			) -> Result<<TestAssets as Inspect<TestAccountId>>::Balance, XcmError> {
+				Ok(AMOUNT)
+			}
+		}
+		impl TakeRevenue for FeeChargerAssetsHandleRefund {
+			fn take_revenue(_: MultiAsset) {}
+		}
+
+		// create new instance
+		type Trader = TakeFirstAssetTrader<
+			TestAccountId,
+			FeeChargerAssetsHandleRefund,
+			TestAssets,
+			TestAssets,
+			FeeChargerAssetsHandleRefund,
+		>;
+		let mut trader = <Trader as WeightTrader>::new();
+
+		// prepare test data
+		let asset: MultiAsset = (Here, AMOUNT).into();
+		let payment = Assets::from(asset.clone());
+		let weight_to_buy = Weight::from_parts(1_0000, 0);
```

Comment:
Extra zero?

Reply:
typo, change to `Weight::from_parts(1_000, 1_000)`